xbmc/cores/dvdplayer/DVDCodecs/Video/DXVA.cpp

```diff
@@ -509,17 +511,26 @@ bool CDXVAContext::CreateSurfaces(D3D11_VIDEO_DECODER_DESC format, unsigned int
   vdovDesc.Texture2D.ArraySlice = 0;
   vdovDesc.ViewDimension = D3D11_VDOV_DIMENSION_TEXTURE2D;
 
-  for (unsigned i = 0; i < count; ++i)
+  size_t i;
+  for (i = 0; i < count; ++i)
   {
     vdovDesc.Texture2D.ArraySlice = D3D11CalcSubresource(0, i, texDesc.MipLevels);
-    if (FAILED(m_service->CreateVideoDecoderOutputView(*texture, &vdovDesc, &surfaces[i])))
+    hr = m_service->CreateVideoDecoderOutputView(texture, &vdovDesc, &surfaces[i]);
+    if (FAILED(hr))
     {
-      CLog::Log(LOGNOTICE, "%s - failed creating surfaces", __FUNCTION__);
-      return false;
+      CLog::Log(LOGERROR, "%s - failed creating surfaces", __FUNCTION__);
+      break;
     }
+  }
+  SAFE_RELEASE(texture);
 
+  if (FAILED(hr))
+  {
+    for (size_t j; j < i; ++j)
```

Comment:
nice catch
j should be less than i, because if error occurs then resource at i is not created and has a nullptr or bad value and destroying it may cause a segfault.